### PR TITLE
Fix compile error from change to JavaScriptExecutorFactory constructor

### LINKED ISSUE #11: constructor JSCJavaScriptExecutorFactory in class JSCJavaScriptExecutorFactory cannot be applied to given types
Hello,
Using this lib with the latest stable RN version 0.51.0 I have encoutered the following issue when trying to run an Android project
```
myproject/node_modules/react-native-threads/android/src/main/java/com/reactlibrary/ReactContextBuilder.java:57: error: constructor JSCJavaScriptExecutorFactory in class JSCJavaScriptExecutorFactory cannot be applied to given types;
        JavaScriptExecutor jsExecutor = new JSCJavaScriptExecutorFactory().create();
                                        ^
  required: String,String
  found: no arguments
  reason: actual and formal argument lists differ in length
1 error
:react-native-threads:compileReleaseJavaWithJavac FAILED
```
I have scoured the list of breaking changes since 0.49 but haven't managed to pinpoint the culprit.
I also haven't had the chance to try this on an IOS device.
Could I  trouble you for a quick fix?

Thank you very much.

### PR DESCRIPTION
**Pull request by newyankeecodeshop**
Fixes #11.

I also upgraded the Android build tool because Android Studio couldn't sync the gradle project with the older tool.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -23,6 +23,6 @@
     "index.js"
   ],
   "peerDependencies": {
-    "react-native": "~0.49.0"
+    "react-native": ">=0.50.0"
```
Comment (newyankeecodeshop) [inline package.json]: Should work with 0.51 too

**Comment by newyankeecodeshop** [comment]
I think a separate PR will be needed for iOS

**Comment by Traviskn** [comment]
@newyankeecodeshop Thanks a ton for this PR!  I'm going to test it out on the example app with react-native 0.50 and 0.51 just to verify.  

The iOS side of things has been pretty stable, just the android API's that keep changing, so I'm pretty sure iOS will still work fine on newer react-native versions.  I'll double check that too though.

Thanks again!

**Comment by Traviskn** [comment]
Working great 👍

**Pull request merged**